### PR TITLE
Add identical_to API for equal treatment of NULL values

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

File: ibis/impala/tests/test_exprs.py [hunk 1]
```diff
@@ -1564,3 +1564,11 @@ def test_unions_with_ctes(self):
 
         expr = join1.union(join2)
         self.con.explain(expr)
+
+    def test_identical_to(self):
```
Comment (wesm) [inline ibis/impala/tests/test_exprs.py]: this test isn't being run because it's inside an ImpalaE2E class

Comment (cpcloud) [inline ibis/impala/tests/test_exprs.py]: cool, i'll move it

**Comment by wesm** [comment]
I can merge this if you rebase. At some point will want to add some correctness tests for the Impala code, perhaps we can leave that as a TODO in a later github issue

**Comment by tdhopper** [comment]
Thanks for all your work on Ibis, @cpcloud!

**Comment by cpcloud** [comment]
@wesm rebased

**Comment by cpcloud** [comment]
I have the impala minimal docker image on my local machine so i can add a test for the correctness here and merge after that passes.

**Comment by wesm** [comment]
Can you rebase? Sorry... =/

**Comment by cpcloud** [comment]
Rebasing this. Moving this to 0.10, as this isn't critical and I'd like to write some more test cases.

**Comment by cpcloud** [comment]
Looks like impala implemented `IS DISTINCT FROM` since I put this up. I'll refactor and rebase and push it up to get it merged.

**Comment by cpcloud** [comment]
Merging on pass.

**Pull request closed without merging**